src/components/stateful-enroll/data/hooks/useStatefulEnroll.js

```diff
@@ -32,9 +34,16 @@ const useStatefulEnroll = ({
   onSuccess,
   onError,
   onBeginRedeem,
+  courseEnrollmentsByStatus,
 }) => {
   const [transaction, setTransaction] = useState();
-
+  const optimizelyHandler = useOptimizelyEnrollmentClickHandler(
+    contentKey,
+    courseEnrollmentsByStatus,
+  );
+  const searchHandler = useTrackSearchConversionClickHandler({
+    eventName: EVENT_NAMES.clickedToEnrollPage,
```

Comment:
nit: we may want to consider co-locating these `EVENT_NAMES` with the `EVENT_NAMES` that exists in the course constants here: https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/a5754ba055311a73bcaa0d60c489bea7748af27a/src/components/course/data/constants.js#L60-L62